src/sentry/utils/suspect_resolutions/get_suspect_resolutions.py

```diff
@@ -1,13 +1,21 @@
 from typing import Sequence
 
 from sentry.models import Activity, Group, GroupStatus
+from sentry.signals import issue_resolved
+from sentry.tasks.base import instrumented_task
 from sentry.utils.suspect_resolutions import analytics
 from sentry.utils.suspect_resolutions.commit_correlation import is_issue_commit_correlated
 from sentry.utils.suspect_resolutions.metric_correlation import is_issue_error_rate_correlated
 
 
-def get_suspect_resolutions(resolved_issue: Group) -> Sequence[Group]:
+@issue_resolved.connect(weak=False)
+def record_suspect_resolutions(organization_id, project, group, user, resolution_type, **kwargs):
+    get_suspect_resolutions.delay(group.id)
```

Comment:
We should put this behind a feature flag so that we can disable it easily if necessary. I think that at the moment this will be a pretty expensive operation for most projects.

Reply:
Just raising this again - we definitely need a feature flag here so that we can disable if we see problems.